src/emit/substrate/mod.rs

```diff
@@ -22,6 +22,109 @@ pub(super) mod target;
 // When using the seal api, we use our own scratch buffer.
 const SCRATCH_SIZE: u32 = 32 * 1024;
 
+#[macro_export]
+macro_rules! emit_context {
+    ($binary:expr) => {
+        #[allow(unused_macros)]
+        macro_rules! byte_ptr {
+            () => {
+                $binary.context.i8_type().ptr_type(AddressSpace::Generic)
+            };
+        }
+
+        #[allow(unused_macros)]
+        macro_rules! cnst {
+            ($of_type:ident, $val:expr) => {
+                $binary.context.$of_type().const_int($val, false)
+            };
+        }
+
+        #[allow(unused_macros)]
+        macro_rules! i32_cnst {
+            ($val:expr) => {
+                cnst!(i32_type, $val)
+            };
+        }
+
+        #[allow(unused_macros)]
+        macro_rules! i32_null {
```

Comment:
I think the name `i32_null` is confusing, it sounds like a pointer. How about `i32_zero`?

Reply:
ACK